src/log/format.cr

```diff
@@ -118,7 +118,7 @@ class Log
     # It doesn't write any output if the context is empty.
     # Parameters `before` and `after` can be provided to be written around
     # the value.
-    def context(*, before = nil, after = nil)
+    def context(*, before : _ = nil, after : _ = nil) : Nil
```

Comment:
I believe we're on the side of preferring no type restriction instead of `_` or `forall` if the type can't be restricted. See #15334

Reply:
If the type is really unrestricted - like in this case - I think it makes a lot of sense to explicate that in the type restriction `_`.
Otherwise, it would be indiscernible from an unknown type restriction.

Reply:
That is also my preference, especially from a documentation perspective, but I know others on the team disagreed.

Another alternative proposed in that issue:

```cr
def context(*, before : B = nil, after : A = nil) : Nil forall A, B
end
```

Reply:
`_` is more concise than `forall`, so I'd use that.